iceoryx2-cli/tests/config_description_tests.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2025 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+mod config_descriptions_tests {
+    use std::collections::HashSet;
+    use toml::Value;
+
+    use iceoryx2::config::Config;
+    use iceoryx2_bb_testing::assert_that;
+    use iceoryx2_cli::config_descriptions::get_sections;
+
+    // Recursively walk through toml::Value to flatten all keys into a HashSet
+    fn collect_keys(value: &Value, prefix: String, keys: &mut HashSet<String>) {
+        match value {
+            Value::Table(table) => {
+                for (k, v) in table {
+                    let new_prefix = if prefix.is_empty() {
+                        k.to_string()
+                    } else {
+                        format!("{}.{}", prefix, k)
+                    };
+                    collect_keys(v, new_prefix, keys);
+                }
+            }
+            _ => {
+                keys.insert(prefix);
+            }
+        }
+    }
+
+    #[test]
+    fn check_config_description_is_present_for_all_config_parameters() {
+        let config = Config::default();
+        let toml_string = toml::to_string(&config).expect("Failed to serialize config to TOML");
```

Comment:
Maybe TOML is not the best format for this check. Type like `Option<Foo>` will not show up if the value is `None`. Maybe [RON](https://crates.io/crates/ron) is an alternative